### ISSUE TITLE
When should the window be created hidden?

### ISSUE THREAD

**Issue by marc2332** [AccessKit/accesskit#255]
Hey!

While I was working on accessibility support for [Freya](https://github.com/marc2332/freya) I noticed a weird behavior. The initial redraw request event never happens. So after some digging, I found out it only occurs when creating the Winit adapter

I have only tested this on Windows 11.

When creating the Winit adapter:
![image](https://github.com/AccessKit/accesskit/assets/38158676/6e837599-dfab-46b8-bd71-d20e215462a6)

When not creating it :
![image](https://github.com/AccessKit/accesskit/assets/38158676/7f80b368-4b15-4c04-b40a-67aa4c49f952)

I made a little [reproduction](https://github.com/marc2332/bug-accesskit-winit-initial-redraw-request) example, the important line is [this one](https://github.com/marc2332/bug-accesskit-winit-initial-redraw-request/blob/876a069806dd87776476ac6c593533977e1422d6/src/main.rs#L161)

This issue is easy to fix by simply listening to the Init event of the eventloop and forcing a redraw request:

```rust
 Event::NewEvents(StartCause::Init) => {
    window.request_redraw();
}
```

But, is still not the expected behavior and I wonder if it could be causing other problems I am not aware of.

Anyway, nice job with accesskit 👏

**Comment by mwcampbell** [comment]
Thanks for the report. Can you please specify which platform(s) this is happening on?

**Comment by marc2332** [comment]
> Thanks for the report. Can you please specify which platform(s) this is happening on?

I have only tested this in Windows 11, I can try in Linux if you want, but not MacOS as I don't have a Mac 😄

**Comment by mwcampbell** [comment]
You need to create the window invisible, then create the adapter, then show the window. I'm sorry I didn't make that clear in the accesskit_winit docs. This is crucial on Windows, and possibly important on macOS as well.

**Comment by marc2332** [comment]
> You need to create the window invisible, then create the adapter, then show the window. I'm sorry I didn't make that clear in the accesskit_winit docs. This is crucial on Windows, and possibly important on macOS as well.

Ohh, I will try that, Linux works btw :smile:

**Comment by DataTriny** [comment]
Maybe we should panic if `Window::is_visible() == Some(true)` when initializing the adapter.

**Comment by marc2332** [comment]
> You need to create the window invisible, then create the adapter, then show the window. I'm sorry I didn't make that clear in the accesskit_winit docs. This is crucial on Windows, and possibly important on macOS as well.

Yeah, that works ! ✨ You can close this issue if you want now 😄 But, just as you were suggesting, maybe it could be clarified in the docs/examples.

**Comment by DataTriny** [comment]
Coming back to this, when exactly is it necessary to create the window hidden so that AccessKit can initialize properly?

- [ ] `accesskit_macos`/`Adapter`
- [ ] `accesskit_macos`/`SubclassingAdapter`
- [x] `accesskit_unix`/`Adapter`: N/A
- [ ] `accesskit_windows`/`Adapter`: our example does it, but if I add `WS_VISIBLE` and remove the call to `ShowWindow` it seem to work without issues (tested with NVDA and Narrator on Windows 10). Can it be a timing issue under specific circumstances? If no, then we should document that `UiaInitMarker` must be created when handling `WM_NCCREATE`.
- [x] `accesskit_windows`/`SubclassingAdapter`: window must be hidden.

Once the above items are resolved, we should add a platform-specific note to `accesskit_winit::Adapter::new`.

**Comment by DataTriny** [comment]
Solved by #278
